src/pages/settings/Profile/TimezoneSelectPage.js

```diff
@@ -34,27 +34,48 @@ class TimezoneSelectPage extends Component {
         this.saveSelectedTimezone = this.saveSelectedTimezone.bind(this);
         this.filterShownTimezones = this.filterShownTimezones.bind(this);
 
-        this.currentSelectedTimezone = lodashGet(props.currentUserPersonalDetails, 'timezone.selected', CONST.DEFAULT_TIME_ZONE.selected);
+        this.timezone = lodashGet(props.currentUserPersonalDetails, 'timezone', CONST.DEFAULT_TIME_ZONE);
         this.allTimezones = _.chain(moment.tz.names())
             .filter(timezone => !timezone.startsWith('Etc/GMT'))
             .map(timezone => ({
                 text: timezone,
                 keyForList: timezone,
 
                 // Include the green checkmark icon to indicate the currently selected value
-                customIcon: timezone === this.currentSelectedTimezone ? greenCheckmark : undefined,
+                customIcon: timezone === this.timezone.selected ? greenCheckmark : undefined,
 
                 // This property will make the currently selected value have bold text
-                boldStyle: timezone === this.currentSelectedTimezone,
+                boldStyle: timezone === this.timezone.selected,
             }))
             .value();
 
         this.state = {
-            timezoneInputText: this.currentSelectedTimezone,
+            timezoneInputText: this.timezone.selected,
             timezoneOptions: this.allTimezones,
         };
     }
 
+    componentDidUpdate() {
+        // Update timezoneInputText & all timezoneOptions when the timezone object changes
+        const newTimezone = lodashGet(this.props.currentUserPersonalDetails, 'timezone', CONST.DEFAULT_TIME_ZONE);
+        if (_.isEqual(this.timezone, newTimezone)) { return; }
+        this.timezone = newTimezone;
+        const updatedAllTimezones = _.map(this.allTimezones, timezone => ({
+            ...timezone,
+
+            // Include the green checkmark icon to indicate the currently selected value
+            customIcon: timezone === this.timezone.selected ? greenCheckmark : undefined,
+
+            // This property will make the currently selected value have bold text
+            boldStyle: timezone === this.timezone.selected,
+        }));
+
+        this.setState({
+            timezoneInputText: this.timezone.selected,
+            timezoneOptions: updatedAllTimezones,
+        });
+    }
```

Comment:
@priyeshshah11  Why do we need this `componentDidUpdate`?

Reply:
>It also adds componentDidUpdate in order to update the timezone options when automatic is toggled on/off as navigating back doesn't unmount the page, thus it won't update the timezone options & stay disabled without this.

We should put that instead of _// Update timezoneInputText & all timezoneOptions when the timezone object changes_

Reply:
@mollfpr 
>It also adds componentDidUpdate in order to update the timezone options when automatic is toggled on/off as navigating back doesn't unmount the page, thus it won't update the timezone options & stay disabled without this.

Reply:
Done

Reply:
Thanks @priyeshshah11 that will help to explain what's going on 👍